tests/certification/bindings/azure/eventhubs/eventhubs_test.go

```diff
@@ -334,61 +347,73 @@ func TestEventhubBindingMultiplePartition(t *testing.T) {
 	consumerGroup1 := watcher.NewUnordered()
 	consumerGroup2 := watcher.NewUnordered()
 
-	sendAndReceive := func(ctx flow.Context) error {
+	metadata0 := map[string]string{
+		messageKey: partition0,
+	}
 
-		client, err := dapr.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
-		require.NoError(t, err, "dapr init failed")
+	metadata1 := map[string]string{
+		messageKey: partition1,
+	}
+	sendAndReceive := func(metadata0 map[string]string, metadata1 map[string]string) flow.Runnable {
+		return func(ctx flow.Context) error {
+			client, err := dapr.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
+			require.NoError(t, err, "dapr init failed")
 
-		// Define what is expected
-		outputmsg := make([]string, numMessages)
-		for i := 0; i < numMessages; i++ {
-			outputmsg[i] = fmt.Sprintf("output binding: Message %03d", i)
-		}
-		consumerGroup1.ExpectStrings(outputmsg...)
-		time.Sleep(40 * time.Second)
-
-		// Send events from output binding
-		for _, msg := range outputmsg {
-			ctx.Logf("Sending eventhub message: %q", msg)
-
-			err := client.InvokeOutputBinding(
-				ctx, &dapr.InvokeBindingRequest{
-					Name:      "azure-partition0-binding",
-					Operation: "create",
-					Data:      []byte(msg),
-				})
-			require.NoError(ctx, err, "error publishing message")
-		}
+			// Define what is expected
+			outputmsg := make([]string, 50)
+			for i := 0; i < 50; i++ {
+				outputmsg[i] = fmt.Sprintf("output binding: Message %d, partitionkey: %s", i, metadata0[messageKey])
+			}
+			consumerGroup1.ExpectStrings(outputmsg...)
+			time.Sleep(40 * time.Second)
 
-		// Assert the observed messages
-		consumerGroup1.Assert(ctx, time.Minute)
+			// Send events from output binding
+			for _, msg := range outputmsg {
+				ctx.Logf("Sending eventhub message: %q", msg)
 
-		// Define what is expected
-		outputmsg2 := make([]string, numMessages)
-		for i := 0; i < numMessages; i++ {
-			outputmsg2[i] = fmt.Sprintf("output binding: Message %03d", i)
-		}
-		consumerGroup2.ExpectStrings(outputmsg2...)
-		time.Sleep(40 * time.Second)
-
-		// Send events from output binding
-		for _, msg2 := range outputmsg2 {
-			ctx.Logf("Sending eventhub message: %q", msg2)
-
-			err := client.InvokeOutputBinding(
-				ctx, &dapr.InvokeBindingRequest{
-					Name:      "azure-partition1-binding",
-					Operation: "create",
-					Data:      []byte(msg2),
-				})
-			require.NoError(ctx, err, "error publishing message")
-		}
+				err := client.InvokeOutputBinding(
+					ctx, &dapr.InvokeBindingRequest{
+						Name:      "azure-partition0-binding",
+						Operation: "create",
+						Data:      []byte(msg),
+						Metadata:  metadata0,
+					})
+				require.NoError(ctx, err, "error publishing message")
+			}
 
-		// Assert the observed messages
-		consumerGroup2.Assert(ctx, time.Minute)
-		return nil
-	}
+			// Assert the observed messages
```

Comment:
Please immediately send the events to the other binding here first before deleting or asserting anything!

The point is we want to see that there is potential cross-pollination so to say. There should not be any and that's what we must verify.